cmd/cli/config/set.go

```diff
@@ -4,164 +4,98 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"reflect"
 	"strconv"
-	"strings"
 	"time"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
-	"golang.org/x/exp/slices"
-	"gopkg.in/yaml.v3"
 
 	"github.com/bacalhau-project/bacalhau/cmd/util"
 	"github.com/bacalhau-project/bacalhau/cmd/util/hook"
-	"github.com/bacalhau-project/bacalhau/cmd/util/parse"
-	"github.com/bacalhau-project/bacalhau/pkg/config"
 	"github.com/bacalhau-project/bacalhau/pkg/config/types"
-	"github.com/bacalhau-project/bacalhau/pkg/logger"
-	"github.com/bacalhau-project/bacalhau/pkg/models"
 )
 
 func newSetCmd() *cobra.Command {
 	showCmd := &cobra.Command{
-		Use:      "set",
-		Args:     cobra.MinimumNArgs(2),
-		Short:    "Set a value in the config.",
-		PreRunE:  hook.ClientPreRunHooks,
-		PostRunE: hook.ClientPostRunHooks,
+		Use:          "set",
+		Args:         cobra.MinimumNArgs(2),
+		Short:        "Set a value in the config.",
+		PreRunE:      hook.ClientPreRunHooks,
+		PostRunE:     hook.ClientPostRunHooks,
+		SilenceUsage: true,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			// initialize a new or open an existing repo. We need to ensure a repo
 			// exists before we can create or modify a config file in it.
 			_, err := util.SetupRepoConfig(cmd)
 			if err != nil {
 				return fmt.Errorf("failed to setup repo: %w", err)
 			}
-			return setConfig(args[0], args[1:]...)
+			return setConfig(cmd.PersistentFlags().Lookup("path").Value.String(), args[0], args[1:]...)
 		},
 	}
+	cfgDir, err := os.UserConfigDir()
+	if err != nil {
+		cfgDir = ""
+	}
+	configFilePath := filepath.Join(cfgDir, "bacalhau", "config.yaml")
+	showCmd.PersistentFlags().String("path", configFilePath, "Optionally provide a path to a config file to use")
```

Comment:
The `--config` flag accepts either a config file path or a key value pair. This flag (`--path`) only accepts a config file path, and to avoid ambiguity I used `--path` to indicate the flag must accept only a path to a config file as the description of the flag states.

Reply:
yeah more confusing if we use `path`. Lets keep `--config` but make it accept a config file path only

Reply:
done.